test/e2e/install/install_test.go

```diff
@@ -300,3 +329,30 @@ func getDaemonSetByOwnerReference(uid types.UID) (*v1.DaemonSet, error) {
 	}
 	return nil, fmt.Errorf("failed to get daemonset with owner reference uid: %s", uid)
 }
+
+func getCRD(crdName string) (*apiextensionv1.CustomResourceDefinition, error) {
+	crd := &apiextensionv1.CustomResourceDefinition{}
+	key := client.ObjectKey{
+		Name: crdName,
+	}
+	err := e2eclient.Client.Get(context.TODO(), key, crd)
+	return crd, err
+}
+
+func getWorkerNodes() ([]*corev1.Node, error) {
+	nodeList := &corev1.NodeList{}
+	var ret []*corev1.Node
+
+	if err := e2eclient.Client.List(context.TODO(), nodeList); err != nil {
+		return nil, fmt.Errorf("failed to get nodes: %w", err)
+	}
+
+	for _, node := range nodeList.Items {
+		_, ok := node.Labels["node-role.kubernetes.io/worker"]
```

Comment:
you can filter by labels using `ListOptions`.
This code is wrong because is too generic: we need to fetch the nodes referred by the `NodeGroups` we configured.
There is a fair bit of refactoring/code movement that can be useful for tests like this